### DISCUSSION TITLE
what's the correct way to configure hue to access kyuubi

### DISCUSSION THREAD

**Discussion by preetsindhal** [apache/kyuubi#3518] [Q&A]
I have integrated hue with kyuubi by referring to the following documentation
https://kyuubi.apache.org/docs/latest/client/bi_tools/hue.html

I am getting following errors
when i navigate two sparksql from hue UI i see this error:

[19/Sep/2022 03:02:40 -0700] hive_server2_lib INFO     Opening sparksql thrift session for user psingh
[19/Sep/2022 03:02:40 -0700] decorators   ERROR    Error running create_session
Traceback (most recent call last):
  File "/usr/share/hue/desktop/libs/notebook/src/notebook/decorators.py", line 119, in wrapper
    return f(*args, **kwargs)
  File "/usr/share/hue/desktop/libs/notebook/src/notebook/api.py", line 115, in create_session
    response['session'] = get_api(request, session).create_session(lang=session['type'], properties=properties)
  File "/usr/share/hue/desktop/libs/notebook/src/notebook/connectors/hiveserver2.py", line 99, in decorator
    return func(*args, **kwargs)
  File "/usr/share/hue/desktop/libs/notebook/src/notebook/connectors/hiveserver2.py", line 211, in create_session
    'id': session.id
AttributeError: 'NoneType' object has no attribute 'id'

And when I try to run the query i get the following error

beeswax.server.dbms.QueryServerException: Bad status for request TOpenSessionReq(client_protocol=6, username='hue', password=None, configuration={'hive.server2.proxy.user': 'myuser'}):
TOpenSessionResp(status=TStatus(statusCode=3, infoMessages=['*org.apache.kyuubi.KyuubiSQLException:Failed to validate proxy privilege of hue for myuser:15:14', 'org.apache.kyuubi.KyuubiSQLException$:apply:KyuubiSQLException.scala:69', 'org.apache.kyuubi.service.authentication.KyuubiAuthenticationFactory$:verifyProxyAccess:KyuubiAuthenticationFactory.scala:175', 'org.apache.kyuubi.service.TFrontendService:getProxyUser:TFrontendService.scala:134', 'org.apache.kyuubi.service.TFrontendService:getUserName:TFrontendService.scala:145', 'org.apache.kyuubi.service.TFrontendService:getSessionHandle:TFrontendService.scala:160', 'org.apache.kyuubi.server.KyuubiTBinaryFrontendService:OpenSession:KyuubiTBinaryFrontendService.scala:81', 'org.apache.hive.service.rpc.thrift.TCLIService$Processor$OpenSession:getResult:TCLIService.java:1497', 'org.apache.hive.service.rpc.thrift.TCLIService$Processor$OpenSession:getResult:TCLIService.java:1482', 'org.apache.thrift.ProcessFunction:process:ProcessFunction.java:39', 'org.apache.thrift.TBaseProcessor:process:TBaseProcessor.java:39', 'org.apache.kyuubi.service.authentication.TSetIpAddressProcessor:process:TSetIpAddressProcessor.scala:36', 'org.apache.thrift.server.TThreadPoolServer$WorkerProcess:run:TThreadPoolServer.java:286', 'java.util.concurrent.ThreadPoolExecutor:runWorker:ThreadPoolExecutor.java:1149', 'java.util.concurrent.ThreadPoolExecutor$Worker:run:ThreadPoolExecutor.java:624', 'java.lang.Thread:run:Thread.java:750', '*org.apache.hadoop.security.authorize.AuthorizationException:User: hue is not allowed to impersonate myuser:0:-1'], sqlState=None, errorCode=0, errorMessage='Failed to validate proxy privilege of hue for myuser'), serverProtocolVersion=6, sessionHandle=None, configuration=None)

thanks.

**Comment by pan3793** [comment]
The key error message is `hue is not allowed to impersonate myuser`, please check your hadoop conf used by Kyuubi
